src/makeCalculator.js

```diff
@@ -4,7 +4,57 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  return {
+    add(num) {
+      result += num;
+
+      return result;
+    },
+
+    subtract(num) {
+      result -= num;
+
+      return result;
+    },
+
+    multiply(num) {
+      result *= num;
+
+      return result;
+    },
+
+    divide(num) {
+      if (num !== 0) {
+        result /= num;
+      } else {
+        result = NaN;
+      }
+
+      return result;
+    },
+
+    operate(callback, num) {
+      if (typeof callback === 'function') {
+        result = callback(num);
```

Comment:
The `operate` method should apply the callback function to the current `result` and the `num` parameter. Modify this line to `result = callback(result, num);` to ensure the operation is applied correctly.